test/framework/src/main/java/org/elasticsearch/test/IndexSettingsModule.java

```diff
@@ -65,4 +65,13 @@ public static IndexSettings newIndexSettings(Index index, Settings settings, Set
         }
         return new IndexSettings(metaData, Settings.EMPTY, (idx) -> Regex.simpleMatch(idx, metaData.getIndex().getName()), new IndexScopedSettings(Settings.EMPTY, settingSet));
     }
+
+    public static IndexSettings newIndexSettings(final IndexMetaData indexMetaData, Setting<?>... setting) {
```

Comment:
I needed one that takes an `IndexMetaData` as its first parameter (because I need to update it to set the state to `CLOSED` for the test).  That's the main distinguishing reason to add this method.